src/input.rs

```diff
@@ -347,6 +351,20 @@ mod tests {
                 .next()
                 .is_none()
         );
+
+        // Missing file
+        let dir = tempdir().unwrap();
+        let file_path = dir.path().join("a_missing_file.csv");
+        assert!(!file_path.exists());
+        assert!(read_csv::<Record>(&file_path).is_err());
+        // optional csv's should return empty iterator
+        assert!(
+            read_csv_optional::<Record>(&file_path)
+                .unwrap()
+                .peekable()
+                .peek()
```

Comment:
```suggestion
                .next()
```

Just slightly simpler and consistent with the test above